tensorflow_cloud/gcp.py

```diff
@@ -0,0 +1,297 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import google.auth
+
+
+def get_project_name():
+    # https://google-auth.readthedocs.io/en/latest/reference/google.auth.html
+    _, project_id = google.auth.default()
+    if project_id is None:
+        raise Exception('Could not determine the GCP project id.')
+
+    return project_id
+
+
+def validate_machine_configuration(
+        cpu_cores, memory, accelerator_type, accelerator_count):
+    valid_configurations = _get_valid_machine_configurations()
+    current_config = (
+        cpu_cores, memory, accelerator_type.value, accelerator_count)
+    if current_config not in valid_configurations:
+        raise ValueError(
+            'Invalid machine configuration: cpu_cores:{}, memory:{}, '
+            'accelerator_type:{}, accelerator_count:{}. Please see the '
+            'following AI platform comptibility table for all valid '
+            'configurations: '
+            'https://cloud.google.com/ml-engine/docs/using-gpus#'
+            'compute-engine-machine-types-with-gpu'.format(
+                cpu_cores, memory, str(accelerator_type), accelerator_count))
+
+
+def get_region():
+    return 'us-central1'
+
+
+def get_accelerator_type(accl_type):
+    if accl_type == 'CPU':
+        return 'ACCELERATOR_TYPE_UNSPECIFIED'
+    if accl_type == 'K80':
+        return 'NVIDIA_TESLA_K80'
+    if accl_type == 'P100':
+        return 'NVIDIA_TESLA_P100'
+    if accl_type == 'V100':
+        return 'NVIDIA_TESLA_V100'
+    if accl_type == 'P4':
+        return 'NVIDIA_TESLA_P4'
+    if accl_type == 'T4':
+        return 'NVIDIA_TESLA_T4'
+    else:
+        raise ValueError('Invalid accelerator type.')
+
+
+def get_machine_type(cpu_cores, memory):
```

Comment:
This is also more simply structured as a `dict` and a `get`

Reply:
Done.